### PR TITLE
xivlauncher: add explicit dependency on aria2

### LINKED ISSUE #220065: xivlauncher: closure does not include aria2
Despite being referenced by the aria2c path patching step, xivlauncher's closure (as seen by running `nix path-info -rsSh nixpkgs#xivlauncher`) does not contain aria2, allowing it to be garbage collected if nothing else on the system requires it, breaking patch downloads if that happens.

1. Install xivlauncher into a profile.
2. Verify that patch downloads work.
3. Manually garbage collect the store, or wait until after an automatic collection.
4. Patch downloads now fail with an "path not found" error for aria2c.

Workaround: Install aria2c to a profile explicitly, to ensure that it has a GC root.

(ping @sersorrel @witchof0x20 per bug template instructions)

### Metadata

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.96, NixOS, 22.11 (Raccoon), 22.11.20230228.d70f5cd`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by keysmashes**
###### Description of changes

Fixes #220065

```console
$ nix path-info (which XIVLauncher.Core) -r | rg aria # before
$ nix path-info /nix/store/b8q1n5alr5jm1kqkdx1ijqnd5ca57gm3-XIVLauncher-1.0.3/ -r | rg aria # after
/nix/store/4dhvsc02hin1inhs2a7vwjv0j7sd3wj4-aria2-1.36.0
/nix/store/6dz5bapb1mcwrfqcvx1y74w8jricfcbw-aria2-1.36.0-bin
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/xivlauncher/default.nix [hunk 1]
```diff
@@ -44,6 +44,9 @@ in
         --replace 'exec' 'exec ${steam-run}/bin/steam-run'
     '' + ''
       wrapProgram $out/bin/XIVLauncher.Core --prefix GST_PLUGIN_SYSTEM_PATH_1_0 ":" "$GST_PLUGIN_SYSTEM_PATH_1_0"
+      # the reference to aria2 gets mangled and isn't detectable by nix: https://github.com/NixOS/nixpkgs/issues/220065
```
Comment (lilyinstarlight) [inline pkgs/games/xivlauncher/default.nix:47]: This is a bit of a nit, but could you specifically mention that it's being mangled as UTF-16LE?

**Review by lilyinstarlight — COMMENTED** [review]
Well, I see 5 different instances that this kind of reference fixing is already done in nixpkgs, but there's 4 different naming conventions of the nix-support files between those instances (`depends`, `undetected-runtime-dependencies`, `manual-runtime-dependencies`, `extra-refs`)

It seems `$out/nix-support/depends` is the one that gets used twice, though, so I guess that's the closest we have to a consistent convention for this

I left one nit and otherwise this looks good for merge

File: pkgs/games/xivlauncher/default.nix (hunk 1 above)
Comment (sersorrel) [reply pkgs/games/xivlauncher/default.nix]: done!

**Review by lilyinstarlight — APPROVED** [review]
Diff lgtm. I'll merge when the ofborg build is finished (assuming no issues)

**Pull request merged**